### DISCUSSION TITLE
Question: Different VATs for activities

### DISCUSSION THREAD

**Discussion by nilskretschmer** [kimai/kimai#2165] [General]
**Description**
I want to create an invoice and I need to specify different VAT values for different activities. For example:
Activity 1 is calculated with 16%  VAT
Activity 2 is calculated with 5% VAT

**Problem**
It seems that I can specify only one VAT value for my invoice. Is there a kind of workaround or some setting I can change to achieve my goal?

**Additional context**
 - Kimai version: 1.11.1
 - PHP version: 7.4.x

**Comment by kevinpapst** [comment]
Not possible by default.
Kimai is NOT a full-fledged invoicing solution but a time-tracker with basic invoice features.

Maybe you could achieve it with custom templates, customer-fields and your own calculator. But this is a big maybe...
If you have ideas on how to improve that in Kimai (code wise) please share your ideas!

**Comment by nilskretschmer** [comment]
Hey there,
thank you for your quick response!
I fully understand that Kimai cannot be used for every use case. Anyway it is a very nice software and I like the design, too. But I think this is a kind of model design problem:
In my understanding the VAT should be linked to an activity/product - not the invoice itself. Normally, products/activities require a specific VAT so it would be an improvement to redesign this relationship. I know that this would need some major changes and I'm not sure if this change is even possible. But if you could give me some advice how to start I can try to create such a patch?

Alternatively maybe I can build a plugin which enables VAT for activities and build a custom template and calculator which uses those VAT values for the invoice. If I understood it correctly that's the kind of feature you have in mind?

**Comment by kevinpapst** [comment]
> Normally, products/activities require a specific VAT so it would be an improvement to redesign this relationship.

I have a strong e-commerce background, so I understand where this is coming from.
But if you see it from a time-tracking perspective, where do you have different VATs for different activities? 

> I'm not sure if this change is even possible

I wouldn't go down this road. I heard the request for a second VAT setting for an invoice (which seems to be necessary in some countries) but I canÖt remember ever hearing a request like yours before. So I am pretty sure that almost all users who use the invoice system can live with the way it currently works. Which is definitely not the best solution for everyone - I get that - but it seems to be an acceptable compromise.

But still: you could write your own logic to achieve that. 
Extend activities with custom fields for the VAT rates and then build a custom template. You have to calculate the amounts yourself, which can be done with your own calculator: https://www.kimai.org/documentation/developers.html#invoices

**Reply by vic-t** [reply]
In general, NKnuelle is right, VAT is something that should be defined per entry, not per invoice. 

> But if you see it from a time-tracking perspective, where do you have different VATs for different activities?

At least in theory, VAT rates are not set in stone and the country's lawmakers may decide to change them basically at any time. If this happens mid-project, you may end up with entries that have a different VAT rate. Admittedly, this doesn't happen too often, but it's something to keep in mind. Switzerland, for example, had two changes in the last 10 years and it is already known there will be another one 9 years from now.

Other than that, having different rates for different activities is rather unusual in the services sector, I agree. But you may have customers with a different rate. For example, if serving a customer living abroad, depending on the country another VAT rate or, more commonly, a rate of 0% hast to be applied. 

Sure, you could just create a 0% invoice and use that but that means you have to remember which invoice to use for every customer every time you create one. It would be cleaner and safer to store this information where it belongs. 

In the long term, it may thus make sense to extend the timesheet data schema with a field for the VAT rate and add this field basically everywhere (user, customer, project, activity, time entry). Calculation of the correct VAT rate could follow a similar logic as you currently do with work rates (https://www.kimai.org/documentation/timesheet.html#rate-calculation).

**Comment by kevinpapst** [comment]
Or you use an invoicing software like Grandtotal. You could build another connector, if GT is not working for you:
https://www.kimai.org/store/grandtotal-plugin-for-kimai.html

In the end it doesn't have to be Kimai, if it doesn't fit your use case ...

**Reply by nilskretschmer** [reply]
As I see GT is Mac-only so this won't work for me :) Do you know of any other FOSS invoicing software which is easy to integrate?
